projects/igniteui-angular/src/lib/time-picker/time-picker.component.ts

```diff
@@ -99,473 +107,328 @@ export interface IgxTimePickerValidationFailedEventArgs extends IBaseEventArgs {
         }`
     ]
 })
-export class IgxTimePickerComponent implements
+export class IgxTimePickerComponent extends PickerBaseDirective
+    implements
     IgxTimePickerBase,
     ControlValueAccessor,
-    EditorProvider,
     OnInit,
+    OnChanges,
     OnDestroy,
     AfterViewInit,
-    AfterViewChecked {
-
+    Validator {
     /**
      * An @Input property that sets the value of the `id` attribute.
      * ```html
-     * <igx-time-picker [id]="'igx-time-picker-5'" format="h:mm tt" ></igx-time-picker>
+     * <igx-time-picker [id]="'igx-time-picker-5'" [displayFormat]="h:mm tt" ></igx-time-picker>
      * ```
      */
     @HostBinding('attr.id')
     @Input()
     public id = `igx-time-picker-${NEXT_ID++}`;
 
     /**
-     * An @Input property that allows you to disable the `igx-time-picker` component. By default `disabled` is set to false.
+     * The format used when editable input is not focused. Defaults to the `inputFormat` if not set.
+     *
+     * @remarks
+     * Uses Angular's `DatePipe`.
+     *
+     * @example
      * ```html
-     * <igx-time-picker [disabled]="'true'" [vertical]="true" format="h:mm tt" ></igx-time-picker>
+     * <igx-time-picker displayFormat="mm:ss"></igx-time-picker>
      * ```
+     *
      */
     @Input()
-    public disabled = false;
+    public displayFormat: string;
```

Comment:
Technically, that's inherited already, though just for the documentation it might be okay to keep. Same with a few others.